app/src/lib/stores/app-store.ts

```diff
@@ -3089,7 +3081,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
             },
             gitContext: {
               kind: 'checkout',
-              branchToCheckout: foundBranch.name,
+              branchToCheckout: branch,
```

Comment:
No, we want to avoid using the branch name here because whatever is set in this property is what ends up getting passed to the second `_checkoutBranch` invocation from the error handler. Since `_checkoutBranch` no longer supports string branch names the definition for this property no longer includes the string union type.

https://github.com/desktop/desktop/blob/30305ab02a1efef6404f940e3443a5805c10c41c/app/src/lib/git-error-context.ts#L18